python/tvm/autotvm/task/task.py

```diff
@@ -48,6 +50,105 @@ def _lookup_task(name):
     return task
 
 
+def _get_compute(name):
+    """get compute by given name.
+
+    Parameters
+    ----------
+    name: name of compute
+    """
+    task = _lookup_task(name)
+    return task.fcompute
+
+
+def format_subgraph_task_name(name, args):
+    if re.match(r"\w+\d+$", name):
+        # remove number of reduplicative subgraph name like '_1'
+        name = re.sub(r"[_]\d+$", "", name)
+    str_args = format_args(args).encode("utf-8")
+    hash_key = hashlib.md5(str_args).hexdigest()
+    return name + "_" + hash_key
+
+
+def format_args(args):
+    """format arguments of a topi function to a string
+
+    Parameters
+    ----------
+    args: list of hashable or Tensor
+    """
+
+    def _encode(x):
```

Comment:
What about use `serialize_args(...)` instead? Which will return a tuple, and then try to hash that tuple (as part of subgraph key)

Reply:
have added an annotation in code -> get subgraph's workload which is attached to the last op, because the anchor op has attached workload too, so we should get by post order